src/backend/Csrs.Api/Models/ModelExtensions.cs

```diff
@@ -221,6 +223,27 @@ public static Message ToMessage(MicrosoftDynamicsCRMssgCsrscommunicationmessage
             return value;
         }
 
+        private static async Task<int?> GetLookupValueAsync(IDynamicsClient dynamicsClient, string attributeName, string? label, CancellationToken cancellationToken)
```

Comment:
Why are we doing this, we already had "cached" versions of these.  We should not be looking things up by label. We should be looking up things by Id which is unique.

Reply:
it was redeveloped with using cache